src/Passes/Source/Passes/QirAllocationAnalysis/QirAllocationAnalysis.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Llvm/Llvm.hpp"
+
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class QirAllocationAnalysisAnalytics : public llvm::AnalysisInfoMixin<QirAllocationAnalysisAnalytics>
```

Comment:
nit: The naming here seems a bit redundant; maybe QirAllocationAnalytics is sufficiently descriptive?

Reply:
As such I agree. The reason is that analysis typically comes with two components: The analytics component and the printer.